pkg/policy/mapstate.go

```diff
@@ -292,23 +320,30 @@ func (keys MapState) deleteKeyWithChanges(key Key, owner MapStateOwner, adds, de
 // denyPreferredInsertWithChanges inserts a key and entry into the map by giving preference
 // to deny entries, and L3-only deny entries over L3-L4 allows.
 // Incremental changes performed are recorded in 'adds' and 'deletes', if not nil.
-func (keys MapState) denyPreferredInsertWithChanges(newKey Key, newEntry MapStateEntry, adds, deletes Keys) {
+func (keys MapState) denyPreferredInsertWithChanges(newKey Key, newEntry MapStateEntry, adds, deletes Keys, identities Identities) {
 	allCpy := allKey
 	allCpy.TrafficDirection = newKey.TrafficDirection
 	// If we have a deny "all" we don't accept any kind of map entry
 	if v, ok := keys[allCpy]; ok && v.IsDeny {
 		return
 	}
+	newEntryNets := newEntry.getNets(identities, identity.NumericIdentity(newKey.Identity))
 
 	if newEntry.IsDeny {
-		// case for an existing allow L4-only and we are inserting deny L3-only
 		switch {
+		// case for inserting a deny L3-only
```

Comment:
Would this help clarify?

```suggestion
		// case for inserting a deny L3-only when there are existing allows for the same traffic at L3 or L4
```

Reply:
Actually, due to `(k.DestPort != 0 || k.Nexthdr != 0) `, this is still only handling the case where there is an existing l4-only allow.

Reply:
It is handling the case of a deny L3-only. What it is doing within that case is documented further down.